rfc/src/template.md

```diff
@@ -10,47 +10,66 @@
-- If the RFC is related to architectural changes and there are no visible changes to UX, please state so.
+Please include:
+  - High level user flow description
+  - Any new major functions or attributes that will be added to Kani library.
+  - New command line options or subcommands (no need to mention the unstable flag).
+  - What are possible user errors and how are they presented. E.g: Compilation vs Verification error.
+  - Substantial changes to existing functionality or Kani output.
 
-## Detailed Design
+If the RFC is related to architectural changes and there are no visible changes to UX, please state so.
+No further explanation is needed.
 
-This is the technical portion of the RFC. Please provide high level details of the implementation you have in mind:
+## Software Design
 
-- What are the main components that will be modified? (E.g.: changes to `kani-compiler`, `kani-driver`, metadata,
-  installation...)
-- How will they be modified? Any changes to how these components communicate?
-- Will this require any new dependency?
+**We recommend you to leave this empty for the first version of your RFC**.
+
+This is the technical portion of the RFC. Please provide a high level description your design.
+
+- What are the main components that will be modified? (E.g.: changes to `kani-compiler`, `kani-driver`, metadata, proc-macros, installation...)
+- Will there be changes to the components interface?
+- Any changes to how these components communicate?
 - What corner cases do you anticipate?
 
 ## Rationale and alternatives
 
-- What are the pros and cons of this design?
+This is the section where you discuss the decisions you made.
+
+- What are the pros and cons of the UX? What would be the alternatives?
 - What is the impact of not doing this?
-- What other designs have you considered? Why didn't you choose them?
+- Any pros / cons on how you designed this?
 
 ## Open questions
 
-- Is there any part of the design that you expect to resolve through the RFC process?
-- What kind of user feedback do you expect to gather before stabilization? How will this impact your design?
+List of open questions + an optional link to an issue that capture the work required to address the open question. Capture the details of the open question in the issue, not here.
+
+Example:
+- Is there any use case that isn't handled yet?
+- Is there any part of the UX that still need some improvement?
+
+Make sure all open questions are addressed before stabilization.
+
+## Out of scope / Future Improvements
+
+List of extensions and possible improvements that you predict for this feature that is out of the
+scope of this RFC.
 
-## Future possibilities
+Feel free to add as many items as you want, but please refrain from adding too much detail. If you want to capture your thoughts or start a discussion, please create a feature request. You are welcome to add a link to the new issue here.
```

Comment:
I would maybe even add that this section is optional. Especially someone submitting a first PR should not focus on what comes later.